c_models/src/my_models/input_types/my_input_type.h

```diff
@@ -20,25 +20,31 @@ typedef struct input_type_t {
     REAL my_parameter;
 } input_type_t;
 
+static inline void _input_type_set_input_value(
+		input_t total, input_type_t *input_type) {
+    if (total > input_type->my_parameter) {
+        input_type->multiplicator = 1.0;
+    } else {
+        input_type->multiplicator += 1.0;
+    }
+}
 
-static inline input_t* input_type_get_input_value(
-        input_t* value, input_type_pointer_t input_type,
+static inline input_t *input_type_get_input_value(
+        input_t *restrict value, const input_type_t *input_type,
         uint16_t num_receptors) {
     use(input_type);
     input_t total = 0.0;
     for (uint32_t i = 0; i < num_receptors; i++) {
         total += value[i];
     }
-    if (total > input_type->my_parameter) {
-        input_type->multiplicator = 1.0;
-    } else {
-        input_type->multiplicator += 1.0;
-    }
+
+    _input_type_set_input_value(total, input_type);
```

Comment:
This is cheating somewhat, as it is going beyond what the function is named (get_input_value(...) really shouldn't modify anything, it should just get the value...).  Suggestions for how to better fix it welcome.

Reply:
The only other solution would be to have another function that lets it do something else... or else to rename this to something like input_type_do_state_update which can return the state after updating (which in our common cases won't have a state).  I guess this is showing what you can do with the current interface though.  I don't think it is too bad that it changes things if that is what is needed?

Reply:
Yes - the issue really is that this currently gives a compiler warning, because it's modifying something which in the function definition is now (reasonably) defined as a const.

Reply:
Ah OK; in that case the const might be too restrictive...  I guess a warning is not too bad though I don't know if it will have any further consequences.

Reply:
OK, so from checking the meanings, I suspect that on SpiNNaker changing "constant" things will work, as even things stored in a "constant" area of memory will be changeable.  This *is* bad practice but it doesn't seem to be dangerous, at least not just now.

Reply:
It doesn't appear to have any effect on the results of running the example script that uses this particular input_type, so I'm happy for it to go in as it fixes master for now.

Reply:
(I have changed what I called this function originally to better explain what's happening here...)